### PR TITLE
[FIX] business_requirement_deliverable: Count unpublished deliverables in portal

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa
ping @CarlosRoca13 @victoralmau

### CODE REVIEW

**Review by victoralmau — APPROVED** [review]

File: business_requirement_deliverable/models/business.py [hunk 1]
```diff
@@ -194,9 +194,13 @@ def _compute_dl_total_revenue(self):
         for r in self:
             r.dl_total_revenue = sum(r.deliverable_lines.mapped("price_total"))
 
+    @api.depends_context("website_id")
     def _compute_dl_count(self):
         for r in self:
-            r.dl_count = len(r.deliverable_lines.ids)
+            if "website_id" in self.env.context:
```
Comment (pedrobaeza) [inline business_requirement_deliverable/models/business.py]: I don't see why you need the website here, as this doesn't depend on website nor the field is depending on it.

**Review by CarlosRoca13 — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please check my comment

**Comment by sergio-teruel** [comment]
Why have I done this?
For one business requirement (BR) published in portal you can published all BR deliveries or any, so in BR portal view we van not show all deliveries counter, we only can display the published deliveries but this behaviour only is for portal website, in backed we want count all deliveries...

image

**Comment by pedrobaeza** [comment]
Ok, I get the goal, but I'm afraid the implementation is not correct, as you can have portal without website, so you need to create a new field to be used in this part and not to rely on such context.

**Comment by sergio-teruel** [comment]
Ok... I will re-check so...

**Comment by victoralmau** [comment]
Finally I have added a new field that already filters the records published on the website.
If this change is correct, we will add it in v14 as well.
What do you think about it?

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**